lib/rspec/expectations/handler.rb

```diff
@@ -66,10 +72,16 @@ def self.opposite_should_method
 
     # @private
     class NegativeExpectationHandler
-      def self.handle_matcher(actual, initial_matcher, message=nil, &block)
-        ExpectationHelper.with_matcher(self, initial_matcher, message) do |matcher|
+      def self.handle_matcher(actual, initial_matcher, custom_message=nil, &block)
+        ExpectationHelper.with_matcher(self, initial_matcher, custom_message) do |matcher|
           return ::RSpec::Matchers::BuiltIn::NegativeOperatorMatcher.new(actual) unless initial_matcher
-          does_not_match?(matcher, actual, &block) || ExpectationHelper.handle_failure(matcher, message, :failure_message_when_negated)
+
+          is_not_a_match = does_not_match?(matcher, actual, &block)
```

Comment:
If you change above then I'd call this `negated_match_result`

Reply:
Yep good idea! Much better like this.